portfolio/src/main/webapp/blog.html

```diff
@@ -7,9 +7,9 @@
     <link rel="stylesheet" href="style.css">
     <script src="script.js"></script>
 </head>
-<body onload="setColor();loadPosts();">
+<body onload="setColor();loadPosts();checkAuth('blog')">
```

Comment:
Since now you're calling 3 functions here. I would replace it with a single function called `init('blog')`, and define it in `script.js`.